### PR TITLE
EntityCountDashboard with carousel for multiple disagregations

### PR DESCRIPTION
**Pull request by sadaf895**
closes: #2557

- [x] Data loading and grouping.
- [x] UI need to be improved.

### CODE REVIEW

File: src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts [hunk 1]
```diff
@@ -103,14 +104,15 @@ export class EntityCountDashboardComponent
 
   goToChildrenList(filterId: string) {
     const params = {};
-    params[this.groupBy] = filterId;
+    params[this.groupBy[0]] = filterId;
 
-    this.router.navigate([this._entity.route], { queryParams: params });
+    this.router.navigate([Entity.route], { queryParams: params });
```
Comment (sleidig) [inline src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts:109]: did you test the navigation in the app (clicking on one of the lines in the widget (e.g. the "female" children))? I think `Entity.route` will not work.

**Review by sleidig — COMMENTED** [review]
This first step looks good :+1: 

We've merge some other things into master. Can you try to merge the master back into your branch and resolve the conflicts?

File: src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts (hunk 1 above)
Comment (sadaf895) [reply src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts:109]: Yeah! true. I have resolved it

**Comment by sadaf895** [comment]
@sleidig.. Please look into the error it has been occuring after reset packages.

File: src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.html [hunk 2]
```diff
@@ -3,16 +3,55 @@
...
+      display: flex;
+      justify-content: center;
+      align-items: center;
+      padding: 0.5em;
+    "
+  >
+    <button
+      mat-icon-button
+      matTooltip="Previous Grouping"
+      i18n-matTooltip~
```
Comment (sleidig) [inline src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.html:19]:
```suggestion
      i18n-matTooltip
```

File: src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.html [hunk 3]
```diff
@@ -3,16 +3,55 @@
...
+
   <div class="table-wrapper">
     <table
       mat-table
       i18n-aria-label="Label for children count dashboard"
       aria-label="Table showing disaggregation of the beneficiaries"
     >
       <ng-container matColumnDef="label">
-        <td *matCellDef="let group">
+        <td *matCellDef="let group" scope="row">
```
Comment (sleidig) [inline src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.html:54]: what's the meaning of scope="row"?

File: src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts [hunk 4]
```diff
@@ -62,17 +81,20 @@ export class EntityCountDashboardComponent
    *
    * Default is "center".
    */
-  @Input() groupBy = "center";
+  @Input() groupBy: string[] = ["center", "gender"];
```
Comment (sleidig) [inline src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts:84]: We'll need a (very simple) config migration that changes groupBy if it's a string and wraps it as a single item in an array (for existing configs in our client's databases).

@sadaf895 , can you try to write a unit test and then a migration function for this in our ConfigService?

File: src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts [hunk 5]
```diff
@@ -89,29 +111,38 @@ export class EntityCountDashboardComponent
...
-    this.updateCounts(entities.filter((e) => e.isActive));
+    for (const groupByField of this.groupBy) {
+      await this.loadDataForGroupBy(groupByField);
+    }
     this.label = this._entity.labelPlural;
     this.entityIcon = this._entity.icon;
   }
 
+  async loadDataForGroupBy(groupByField: string) {
+    const entities = await this.entityMapper.loadType(this._entity);
```
Comment (sleidig) [inline src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts:122]: we don't have to load the entities for each grouping afresh. Let's move this outside of this function and add a method parameter to pass the same array of entities into each call of `loadDataForGroupBy(groupByField, entities)`

File: src/environments/environment.ts [hunk 6]
```diff
@@ -33,7 +33,7 @@ export const environment = {
   remoteLoggingDsn: undefined, // only set for production mode in environment.prod.ts
 
   demo_mode: true,
-  session_type: SessionType.mock,
+  session_type: SessionType.local,
```
Comment (sleidig) [inline src/environments/environment.ts:36]: before checking into github, revert this (we don't want to merge the SessionType change into our master branch)

**Review by sleidig — CHANGES_REQUESTED** [review]
Nice work! Just a few more details

File: src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts [hunk 7]
```diff
@@ -37,13 +42,27 @@ interface EntityCountDashboardConfig {
     DashboardListWidgetComponent,
     EntityBlockComponent,
     NgIf,
+    MatButtonToggleGroup,
+    CommonModule,
+    FormsModule,
```
Comment (sleidig) [inline src/app/features/dashboard-widgets/entity-count-dashboard-widget/entity-count-dashboard/entity-count-dashboard.component.ts:47]: I think we are now not using these in the final implemention, are we? Let's remove if not required.

**Comment by sleidig** [comment]
> @sleidig.. Please look into the error it has been occuring after reset packages.

I don't see any errors for me. It runs well on my machine.

File: src/app/core/config/config.service.ts [hunk 8]
```diff
@@ -395,3 +396,20 @@ const addDefaultNoteDetailsConfig: ConfigMigration = (key, configPart) => {
 
   return configPart;
 };
+
+const migrateGroupByConfig: ConfigMigration = (key, configPart) => {
+  // Check if we are working with the EntityCountDashboard component and within the 'config' object
+  if (
+    key === "config" &&
```
Comment (sleidig) [inline src/app/core/config/config.service.ts:403]: This is still not fully fool-proof. As I said: You should check if there is a property "component" with value "EntityCountDashboard". Only then you can be certain that the "config" property next to that really is the type of config you are looking to migrate.

**Review by sleidig — APPROVED** [review]
Looks good now!

**Pull request merged**